crates/ruff_cli/src/lib.rs

```diff
@@ -424,7 +424,7 @@ pub fn check(args: CheckCommand, log_level: LogLevel) -> Result<ExitStatus> {
         if cli.statistics {
             printer.write_statistics(&diagnostics, &mut summary_writer)?;
         } else {
-            printer.write_once(&diagnostics, &mut summary_writer)?;
+            printer.write_once(&diagnostics, &mut summary_writer, &pyproject_config)?;
```

Comment:
So one potential issue here is that `pyproject_config` isn't guaranteed to contain the correct set of enabled rules for all files. Ruff supports hierarchical configuration, so you can have different configuration files that apply to different subdirectories in your project. The `pyproject_config` here really just represents the settings in the current working directory.

Could we infer the set of rules from the diagnostic messages? (What is this used for on the SARIF side?)

Reply:
Gotcha--thanks for that context.  SARIF lists the rules applied as part of the tool description so that you can be sure you're getting an apples to apples comparison, rather than just saying both times you checked with a particular rust version.  

While the field is not mandatory in the SARIF specification [Section 3.19.23 rules property](https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/sarif-v2.1.0-errata01-os-complete.html#_Toc141790806) it is [required by Github](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning).   Also see: [Understand Rules and Results](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#understanding-rules-and-results)

I'm not exactly sure how github would handle having the `rules` be supplied as only the ones for which diagnostics could be found as that's not documented.

Reply:
Alternatively, could we just include _all_ rules?

Reply:
Including all rules is probably the safest bet.  In my experience with code scanning on github, I've never looked at what checks were done, only the findings, so this is probably closest to the SARIF expectation.  My gut is that most people have most rules applied, and only turn off a handful, but I could be wrong, especially for older/larger projects. 

Only downsides are if someone complains and then we change it, if others are using it in their CI, the rules set will change, which again, not sure how bad that is and that the file size will be a bit larger.

Reply:
Let's try all-rules for now. You can iterate over them with `Rule::iter()`.